drivers/ethernet/eth_native_posix.c

```diff
@@ -31,19 +31,28 @@
 #include "eth_native_posix_priv.h"
 
 #if defined(CONFIG_NET_L2_ETHERNET)
-#define _ETH_L2_LAYER ETHERNET_L2
-#define _ETH_L2_CTX_TYPE NET_L2_GET_CTX_TYPE(ETHERNET_L2)
 #define _ETH_MTU 1500
 #endif
 
 #define NET_BUF_TIMEOUT MSEC(10)
 
+#if defined(CONFIG_NET_VLAN)
+#define ETH_HDR_LEN sizeof(struct net_eth_vlan_hdr)
+#else
+#define ETH_HDR_LEN sizeof(struct net_eth_hdr)
+#endif
+
+struct ifaces {
+	struct net_if *iface;
+	u16_t vlan_tag;
+};
+
 struct eth_context {
-	u8_t recv[_ETH_MTU + sizeof(struct net_eth_hdr)];
-	u8_t send[_ETH_MTU + sizeof(struct net_eth_hdr)];
+	struct ifaces ifaces[NET_VLAN_MAX_COUNT];
```

Comment:
same as mcus, see my comment there about getting an net_eth_ function that would give you back the right iface from main iface + vtag, so it would avoid book-keeping in driver as well.